examples/open_llama/llama_qlora.json

```diff
@@ -0,0 +1,76 @@
+{
+    "input_model":{
+        "type": "PyTorchModel",
+        "config": {
+            "hf_config": {
+                "model_name": "huggyllama/llama-7b",
+                "task": "text-generation"
+            }
+        }
+    },
+    "data_configs": {
+        "oasst1_train": {
+            "name": "oasst1",
+            "type": "HuggingfaceContainer",
+            "params_config": {
+                "data_name": "timdettmers/openassistant-guanaco",
+                "split": "train",
+                "component_kwargs": {
+                    "pre_process_data": {
+                        "dataset_type": "corpus",
+                        "text_cols": ["text"],
+                        "corpus_strategy": "line-by-line",
+                        "source_max_len": 512,
+                        "pad_to_max_len": false
+                    }
+                }
+            }
+        }
+    },
+    "passes": {
+        "qlora": {
+            "type": "QLoRA",
+            "config": {
+                "compute_dtype": "bfloat16",
+                "quant_type": "nf4",
+                "double_quant": true,
+                "lora_r": 64,
+                "lora_alpha": 16,
+                "lora_dropout": 0.1,
+                "train_data_config": "oasst1_train",
+                "eval_dataset_size": 1024,
+                "training_args": {
+                    "seed": 0,
+                    "data_seed": 42,
+                    "per_device_train_batch_size": 16,
+                    "per_device_eval_batch_size": 16,
+                    "gradient_accumulation_steps": 1,
+                    "gradient_checkpointing": true,
+                    "learning_rate": 0.0002,
+                    "max_steps": 1875,
+                    "logging_steps": 10,
+                    "evaluation_strategy": "steps",
+                    "eval_steps": 187,
+                    "group_by_length": true,
+                    "adam_beta2": 0.999,
+                    "max_grad_norm": 0.3,
+                    "output_dir": "models/checkpoints"
```

Comment:
>output_dir [](http://example.com/codeflow?start=21&length=10)

do we need the output_dir since there is already output_dir in engine?

Reply:
This is for the checkpoints of the training. in case the user wants the checkpoint to be saved. so it's different from the engine's output dir.